rllib/utils/metrics/metrics_logger.py

```diff
@@ -756,21 +755,25 @@ def build_nested_dict(stats_dict, key):
                     and base_stats._clear_on_reduce is False
                 ):
                     for stat in incoming_stats:
-                        base_stats.push(-stat.get_reduce_history()[-2][0])
+                        reduce_by = stat.get_reduce_history()[-2][0]
+                        base_stats.values[-1] -= reduce_by
```

Comment:
I see. Is this because of windows, where we end up pushing the actual value out of the window before reducing it?

Reply:
I think it's because of the way we compute the throughput from the given historic values.
Now that we are doing `Stats._reduced_values()` differently (more often and independent of user's calls to `reduce()`), we should not have negative numbers in the value list anymore (fixed in this PR), but simply have the values list (for a lifetime sum metric) incrementally increase. This way, throughput can always be computed through `(hist[-1] - hist[-2]) / time_delta`.